financial-services-accelerator/consent-mgt-api/org.wso2.financial.services.accelerator.consent.mgt.api.service/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/api/service/ConsentCoreService.java

```diff
@@ -0,0 +1,306 @@
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.financial.services.accelerator.consent.mgt.api.service;
+
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.exceptions.ConsentMgtException;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.AuthorizationResource;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentAttributes;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentHistoryResource;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.ConsentStatusAuditRecord;
+import org.wso2.financial.services.accelerator.consent.mgt.api.dao.models.DetailedConsentResource;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Consent core service interface.
+ */
+public interface ConsentCoreService {
+
+    /**
+     * This method is used to create an authorizable consent. The following functionality contains in this method.
+     * 1. Creates a consent resource
+     * 2. If available, stores consent attributes
+     * 3. Create an audit record for consent creation
+     * 4. If isImplicitAuth parameter is true, creates an authorization resource
+     *
+     * @param consentResource        consent resource
+     * @return returns DetailedConsentResource
+     * @throws ConsentMgtException thrown if any error occur in the process
+     */
+    DetailedConsentResource createConsent(DetailedConsentResource consentResource) throws ConsentMgtException;
+
+    /**
+     * This method is used to get a detailed consent for the provided consent ID. The detailed consent includes
+     * following data if exist in addition to consent resource specific data.
+     * <p>
+     * 1. Relative consent authorization data
+     * 2. Relative consent account mapping data
+     * 3. Relative consent attributes
+     *
+     * @param consentId ID of the consent
+     * @return a detailed consent resource
+     * @throws ConsentMgtException thrown if any error occur in the process
+     */
+    DetailedConsentResource getDetailedConsent(String consentId, String orgId) throws
+            ConsentMgtException;
+
+    /**
+     * This method is used to create an authorization for a consent.
+     * @param consentId consent ID
+     * @param authorizationResource authorization resource
+     * @return returns AuthorizationResource
+     * @throws ConsentMgtException thrown if any error occurs in the process
+     */
+    ArrayList<AuthorizationResource> createConsentAuthorizations(ArrayList<AuthorizationResource> authorizationResource,
+                                                                 String consentId)
```

Comment:
ConsentId should be the first argument. 

```suggestion
    List<AuthorizationResource> createConsentAuthorizations(String consentId, List<AuthorizationResource> authorizationResource)
```

Reply:
resolved in https://github.com/wso2/financial-services-accelerator/pull/600/commits/a6406d8d973f00c253e84310b1d02ca6d62cedd7